internal/rule/rulefunction/library.go

```diff
@@ -1040,7 +1040,7 @@ func LibraryPropertiesArchitecturesFieldSoloAlias() (result ruleresult.Type, out
 		"espressif32":   {"esp32"},
 		"espressif8266": {"esp8266"},
 		"intel_arc32":   {"arc32"},
-		"nordicnrf52":   {"nRF5", "nrf52", "mbed"},
+		"nordicnrf52":   {"nRF5", "nrf52", "mbed", "mbed_edge", "mbed_nano"},
```

Comment:
Just out of curiosity, why `mbed_nano` is under `nordicnrf52`? That core includes both an nrf52 board (the nano 33 ble) and the new nano rp2040 connect which is not an nrf52 one.

Reply:
The idea is that if someone is going to have [the platformIO `nordicnrf52`](https://platformio.org/platforms/nordicnrf52) architecture alias in their library.properties, then they should consider also having one of the equivalent standard Arduino architectures. PlatformIO supports library.properties as a library metadata format, but Arduino boards platforms don't use the PlatformIO platform aliases. Since you can specify multiple compatible architectures in the library.properties `architectures` field, it's often simple to make Arduino libraries support both PlatformIO and the official Arduino development software. Unfortunately, library authors who only use PlatformIO, but advertise their libraries as "Arduino", often don't understand these subtle differences that cause big problems for the users. So this rule is intended to help them to achieve best practices to make their work accessible to everyone.

So this will be a violation of rule `LP045` because it is not truly an Arduino library:
```
architectures=nordicnrf52
```
but this would satisfy `LP045` because here you can consider to be both a PlatformIO library and an Arduino library:
```
architectures=nordicnrf52,mbed_nano
```
and alternatively this will satisfy `LP045`:
```
architectures=nordicnrf52,nrf52
```
and so on...

It's saying that the  `nordicnrf52` architecture could be a valid subset of the `mbed_nano` architecture, not the reverse. So it's OK that not all boards of the `mbed_nano` architecture are nRF52. And just the same it's OK that not all `nordicnrf52`-compatible libraries are mbed compatible because the presence of any of the other standard nRF52-related architectures will satisfy the rule. They only need to have one of the standard architectures of the family in the `architectures` list.

Reply:
Thank you for the exaustive explanation. I wasn't familiar with the mechanism behind the rules and the difference between platformIO specification vs arduino's.